server.go

```diff
@@ -59,23 +58,25 @@ func readKeyData(config *Config) *[]byte {
 			log.Fatalf("ERROR reading keyfile %s: %s!\n", config.KeyPath, err)
 		}
 	}
-	pemBlock, _ := pem.Decode(keyData)
-	if pemBlock != nil {
-		if x509.IsEncryptedPEMBlock(pemBlock) {
-			fmt.Println("Decrypting private key with passphrase...")
-			decoded, err := x509.DecryptPEMBlock(pemBlock, []byte(config.Pwd))
-			if err == nil {
-				header := `PRIVATE KEY` // default key type in header
-				matcher := regexp.MustCompile("-----BEGIN (.*)-----")
-				if matches := matcher.FindSubmatch(keyData); len(matches) > 1 {
-					header = string(matches[1])
-				}
-				keyData = pem.EncodeToMemory(
-					&pem.Block{Type: header, Bytes: decoded})
-			} else {
-				fmt.Printf("Error decrypting PEM-encoded secret: %s\n", err)
-			}
-		}
+
+	passphrase := []byte(config.Pwd)
+	var privateKey interface{}
+	fmt.Println("Decrypting private key with passphrase...")
+	privateKey, err = sshkeys.ParseEncryptedRawPrivateKey(keyData, passphrase)
```

Comment:
Note that if `passphrase` is empty, this will still work provided the key isn't actually encrypted